### PR TITLE
feat(iam): add iam ssh key resource

### PR DESCRIPTION
**Pull request by yfodil**

### CODE REVIEW

File: scaleway/provider.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func Provider(config *ProviderConfig) plugin.ProviderFunc {
 				"scaleway_container_namespace":                 resourceScalewayContainerNamespace(),
 				"scaleway_domain_record":                       resourceScalewayDomainRecord(),
 				"scaleway_domain_zone":                         resourceScalewayDomainZone(),
+				"scaleway_flexible_ip":                         resourceScalewayFlexibleIP(),
```
Comment (remyleone) [inline scaleway/provider.go:70]: This does not have anything to do with iam. Clean up your commits :)

**Review by remyleone — CHANGES_REQUESTED** [review]

File: scaleway/resource_iam_ssh_key.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+				Description: "The date and time of the creation of the iam SSH Key",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the iam SSH Key",
+			},
+			"organization_id": organizationIDSchema(),
+			"project_id":      projectIDSchema(),
+			"disabled": {
```
Comment (Codelax) [inline scaleway/resource_iam_ssh_key.go:54]: `disabled` is optional but is not used in resource

File: scaleway/resource_iam_ssh_key.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		ProjectID: (d.Get("project_id")).(string),
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if _, disabledExists := d.GetOk("disabled"); disabledExists {
+		_, err = iamAPI.UpdateSSHKey(&iam.UpdateSSHKeyRequest{
+			SSHKeyID: d.Id(),
+			Disabled: expandBoolPtr(d.Get("disabled")),
```
Comment (Monitob) [inline scaleway/resource_iam_ssh_key.go:76]: please use the helper `getBool`

File: scaleway/resource_iam_ssh_key.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+				Description: "The date and time of the creation of the iam SSH Key",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the iam SSH Key",
+			},
+			"organization_id": organizationIDSchema(),
+			"project_id":      projectIDSchema(),
+			"disabled": {
```
Comment (Monitob) [inline scaleway/resource_iam_ssh_key.go:51]: Add a test to set and update this attribute

File: scaleway/resource_iam_ssh_key.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+		UpdateContext: resourceScalewayIamSSKKeyUpdate,
+		DeleteContext: resourceScalewayIamSSKKeyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
```
Comment (Monitob) [inline scaleway/resource_iam_ssh_key.go:26]: If optional please consider generating a name

File: scaleway/resource_iam_ssh_key.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+			return diag.FromErr(err)
+		}
+	}
+
+	d.SetId(res.ID)
+
+	return resourceScalewayIamSSHKeyRead(ctx, d, meta)
+}
+
+func resourceScalewayIamSSHKeyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (Monitob) [inline scaleway/resource_iam_ssh_key.go:88]: Many attributes are missing from the schema. Should the `Fingerprint` be exposed?

File: scaleway/resource_iam_ssh_key.go [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+	iamAPI := iamAPI(meta)
+
+	res, err := iamAPI.GetSSHKey(&iam.GetSSHKeyRequest{
```
Comment (Monitob) [inline scaleway/resource_iam_ssh_key.go:91]: Please test the terraform import

File: scaleway/resource_iam_ssh_key.go [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	hasUpdated := false
+
+	if d.HasChange("name") {
+		hasUpdated = true
+		req.Name = expandStringPtr(d.Get("name"))
+	}
+
+	if d.HasChange("disabled") {
```
Comment (Monitob) [inline scaleway/resource_iam_ssh_key.go:124]: Please add a test when the value is null and `true` `false`

**Review by Monitob — CHANGES_REQUESTED** [review]
